sdks/java/core/src/test/java/org/apache/beam/sdk/io/AvroIOTest.java

```diff
@@ -275,6 +290,119 @@ public void testAvroIOWriteAndReadSchemaUpgrade() throws Throwable {
     p.run();
   }
 
+  private TimestampedValue<GenericClass> newValue(GenericClass element, Duration duration) {
+    return TimestampedValue.of(element, new Instant(0).plus(duration));
+  }
+
+  private static class WindowedFilenamePolicy extends FilenamePolicy {
+    String outputFilePrefix;
+
+    WindowedFilenamePolicy(String outputFilePrefix) {
+      this.outputFilePrefix = outputFilePrefix;
+    }
+
+    @Override
+    public ValueProvider<String> getBaseOutputFilenameProvider() {
+      return StaticValueProvider.of(outputFilePrefix);
+    }
+
+    @Override
+    public String apply(Context input) {
+      String filename = outputFilePrefix + "-" + input.getWindow().toString() +  "-"
+          + input.getShardNumber() + "-of-" + (input.getNumShards() - 1) + "-pane-"
```

Comment:
I didn't, because I'm not sure it's the default everyone wants :) If we do want default pattern-based support for this, I'd be in favor of doing it as a followon PR (it's backwards compatible, because it can always be a new Policy object